Hyperopt loss (#1726)
NNPDF/nnpdf: doc/sphinx/source/n3fit/hyperopt.rst

   L_{\rm hyperopt} = \chi^{2}_{(1) \rm pdferr} - \chi^{2}_{(2)}

New hyperoptimization metrics with fold and replica statistics
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The combination of :ref:`k-folding <hyperkfolding-label>` and multi-replica experiments

Reviewer: Maybe provide a brief intro in what you mean here
> The combination of :ref:`k-folding <hyperkfolding-label>` and multi-replica experiments

Perhaps something like the following:
> The hyperopt measures discussed above are all based on performing a single replica fit per fold. However one may also wish to run the hyperoptimization algorithm on fits consisting of many replicas per fold. This is a feasible option in ``n3fit``, since it has been optimised to efficiently run many replica fits in parallel on GPU.

Reviewer 2: Done. Thanks!

> opens several the possibilities for the choice of figure of merit. The simplest option would be to minimize

Reviewer: ```suggestion
opens several possibilities for the choice of figure of merit. The simplest option would be to minimize
```

> the average of :math:`\chi2` across both replica and k folds, *i.e.*,

Reviewer: ```suggestion
the average of :math:`\chi^2` across both replica and k folds, *i.e.*,
```

.. math::
    L_{1} = \frac{1}{n_{\rm fold}} \sum_{k=1}^{n_{\rm fold}} \left< \chi^2_{k} \right>_{\rm rep}.

In NNPDF, this hyperoptimisation metrics is selected via the following generic runcard:

.. code-block:: yaml

        dataset_inputs:
        ...

        kfold:
          loss_type: chi2
          replica_statistic: average
          fold_statistic: average
          partitions:
          - datasets:
            ...
          - datasets:
            ...

        parallel_models: true

>         same_trvl_per_replica: true

Reviewer: ```suggestion
```

   L_{\rm hyperopt} = \chi^{2}_{(1) \rm pdferr} - \chi^{2}_{(2)}

New hyperoptimization metrics with fold and replica statistics
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The hyperopt measures discussed above are all based on performing a single replica fit per fold.

> However one may also wish to run the hyperoptimization algorithm on fits consisting of many

Reviewer: ```suggestion
However, one may also wish to run the hyperoptimization algorithm on fits consisting of many
```

replicas per fold. This is a feasible option in ``n3fit``, since it has been optimised to

> efficiently run many replica fits in parallel on GPU.

Reviewer: ```suggestion
efficiently run many replica fits in parallel on GPU.

```
